control-plane/agents/src/bin/core/controller/reconciler/volume/garbage_collector.rs

```diff
@@ -250,6 +258,53 @@ async fn disown_unused_replicas(
     GarbageCollector::squash_results(results)
 }
 
+/// Volume replicas could have been possibly expanded as part of volume resize,
+/// but the volume expansion failed overall leaving behind expanded replicas. So we
+/// bring back the replica size to what is expected by the volume spec.
+#[tracing::instrument(level = "debug", skip(context, volume), fields(volume.uuid = %volume.uuid(), request.reconcile = true))]
+async fn revert_expanded_replicas(
+    volume: &mut OperationGuardArc<VolumeSpec>,
+    context: &PollContext,
+) -> PollResult {
+    let target = context.specs().volume_target_nexus_rsc(volume.as_ref());
+    if let Some(tgt) = target {
+        let nexus_size = tgt.immutable_ref().size;
+        debug_assert_eq!(nexus_size, volume.as_ref().size);
```

Comment:
Let's not crash...
```suggestion
```

Reply:
Is there a better way to definitively fail(during testing) if we are ever in this situation? Since it's debug assert it's harmless.

Reply:
So is the intent here that it will fail during our testing only and not for users as they run debug?

Reply:
> So is the intent here that it will fail during our testing only and not for users as they run debug?

Yes to fail only in testing since users **won't** run debug(at least for all practical purposes).

Reply:
Could you add a comment saying that this won't be executed for a regular run?